Using Deferred Static Generation with Analytics Tools (#30)
LekoArts/portfolio-v2: www/content/writing/2021-11-04--using-deferred-static-generation-with-analytics-tools/index.mdx

        slug: slug,
      },
      defer: !isTopPage,
    })
  })
}
```

As at the time of writing this, DSG is not run in `gatsby develop` you'll need to run `gatsby build` and `gatsby serve` to see it in action now!

You can check if it worked by looking at the build output at the end of `gatsby build`:

```shell
┌ src/templates/blog-post.jsx
│ ├   /popular-article/
│ ├   ...2 more pages available
│ ├ D /my-first-post/
│ └ D ...2 more pages available
└ src/pages/index.jsx
  └   /

  ╭────────────────────────────────────────────────────────────────╮
  │   (SSG) Generated at build time                                │
  │ D (DSG) Deferred static generation - page generated at runtime │
  │ ∞ (SSR) Server-side renders at runtime (uses getServerData)    │
  │ λ (Function) Gatsby function                                   │
  ╰────────────────────────────────────────────────────────────────╯
```

So as expected, the index page is SSG, and three posts are marked as DSG (and the correct ones). Great! 🎉 Time to see in action on Gatsby Cloud. Commit your changes and push it to trigger a build on Gatsby Cloud (if you work on a feature branch you'll need to open a PR to trigger a build).

> Go to some pages that you now should be marked as deferred. The first request might take a bit longer, after that the requests to other DSG pages should be fast.

Reviewer: ```suggestion
Go to some pages that are marked as deferred. The first request might take a bit longer, after that the requests to other DSG pages should be fast.
```
